### DISCUSSION TITLE
Nerd Fonts v3.0.0

### DISCUSSION THREAD

**Discussion by clienart-bmx** [JanDeDobbeleer/oh-my-posh#4078] [Troubleshoot] [answered]
As explained on https://www.nerdfonts.com/cheat-sheet: 

_With Release v3.0.0 the Material Design Icons were updated and moved to new codepoints (reasons for that are in the release notes).
They are still shown here for reference, but are missing in the actual fonts.
To find for example the replacement for nf-mdi-altimeter enter just altimeter in the search box._

I previously had this line in a template:
`"template": " {{if .Root}}\uf504{{else}}\uf503{{end}} {{ .UserName }} ",`
which basically showed an "account" icon for regular users and an "account alert" icon for admins

But with NF v3, these \u codes have been replaced with other glyphs.

I've determined that i need those new \u codes:
`"template": " {{if .Root}}\uf0005{{else}}\uf0004{{end}} {{ .UserName }} ",`
but apparently OMP doesn't support \u codes that are 5 hex digits long ?
Or is it a Windows Terminal issue ?

I'm using Windows Powershell 5.1 inside the latest release of Windows Terminal on Windows 10 22H2, with the latest release of oh-my-posh.

Thanks in advance for your help !

**Comment by clienart-bmx** [comment, marked as answer]
Never mind, I found the answer: all you need to do is to copy the UTF code of the icon from the nerd font cheat sheet page and it will give you 2 \u codes. For example, the account icon f0004 is \udb80\udc04.

**Reply by stamminator** [reply]
This wasn't working for me, and I've just discovered that it's because the go-yaml library, which ohmyposh uses for parsing YAML, [doesn't support UTF-16 surrogate pairs](https://github.com/go-yaml/yaml/issues/279). When I converted my config to JSON, it worked. Here's hoping [the PR](https://github.com/go-yaml/yaml/pull/1029) a wonderful contributor submitted a few months ago to add support gets merged at some point.

Thankfully, pasting the glyph itself into the template string works in both JSON and YAML.

![image](https://github.com/user-attachments/assets/dfdc3a22-864c-4a02-806f-89d360554974)

![ohmyposh UTF-16 glyphs 2](https://github.com/user-attachments/assets/968bad16-ce26-4a89-aaa0-2f169bceed16)

@JanDeDobbeleer It might be worth adding a warning somewhere, perhaps on https://ohmyposh.dev/docs/configuration/general, about this limitation.

EDIT: After some additional research, I've found that YAML supports `\U` (capital) followed by 8 digits for regular (UTF-8) hexadecimal literals inside double-quoted strings, which means that currently, the YAML way to put hex codes that are 5 or more digits long (a.k.a. non-BMP codepoints) into your templates is to use either this method or the glyph itself, but not UTF-16 surrogate pairs. For example, `nf-md-alien`, which is hex code `f089a`, would be `"\U000f089a"` in YAML and `\udb82\udc9a` in JSON.
